hw3/01-characters.js

```diff
@@ -1,4 +1,73 @@
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const getApi = 'https://thronesapi.com/api/v2/Characters';
 
-fetch(url);
+/**
+ * Asynchronous function to fetch data from a specified URL
+ * @param {string} url - The URL from which to fetch data.
+ * @returns {Promise} - A promise that resolves with the fetched JSON data, or rejects with an error
+ */
+const fetchData = async (url) => {
+    let promise = new Promise((resolve, reject) => {
+        fetch(url)
+            // Parse the response as JSON
+            .then((response) => response.json())
+            // Resolve the promise with the JSON data
+            .then((data) => {
+                resolve(data);
+            })
+            .catch((err) => reject(err));
+    });
+    // Wait for the promise to resolve and store the result in the 'data' variable, then returns the data from the function
+    // or returns an error if the promise is rejected
+    try {
+        let data = await promise;
+        return data;
+    } catch (err) {
+        console.log(err);
+    }
+};
+
+// Display characters from API
+async function displayCharacter() {
+    try {
+        const characterData = await fetchData(getApi);
+
+        const characterDetails = document.getElementById('character');
+        const characterRow = document.createElement('div');
+        characterRow.classList.add('row', 'mb-4', 'mt-4', 'character-row');
+
+        characterData.forEach((character) => {
+            // create div for each character
+            const characterImgCol = document.createElement('div');
+            characterImgCol.classList.add(
+                'col-md-3',
+                'col-6',
+                'mb-5',
+                'character-col',
```

Comment:
Minor - Naming is confusing. 'character-col' gives interpretation of it is a column. But instead it is just an div/card. 
Also same goes for  'character-row'. This is specially confusing because it is very similar to bootstrap naming convention.

Reply:
replaced character-row with character-container as its a container of all images.  And changed character-col to character-card as it is a card of each character